### PR TITLE
Split number type into integer and number

### PR DESCRIPTION
**Pull request by junshiguo**
to avoid NumberFormatException

### CODE REVIEW

File: src/main/java/ml/shifu/shifu/container/meta/MetaFactory.java [hunk 1]
```diff
@@ -341,11 +342,11 @@ private static boolean isJsonIngoreField(Field field) {
...
-    static Set<String> filterSet = Sets.newHashSet(new String[] { "NumHiddenLayers", "ActivationFunc",
-            "NumHiddenNodes", "LearningRate", "DropoutRate", "RegularizedConstant", "L1orL2", "MaxDepth",
-            "MinInstancesPerNode", "MinInfoGain", "MaxStatsMemoryMB", "TreeNum", "Impurity", "FeatureSubsetStrategy",
-            "Loss", "LearningDecay", "Propagation", "GBTSampleWithReplacement", "Kernel", "Const", "Gamma",
-            "EnableEarlyStop", "ValidationTolerance", "MaxLeaves", "MaxBatchSplitSize" });
+    static Set<String> filterSet = Sets.newHashSet(new String[] { "NumHiddenLayers", "ActivationFunc", "NumHiddenNodes",
+            "LearningRate", "DropoutRate", "RegularizedConstant", "L1orL2", "MaxDepth", "MinInstancesPerNode",
+            "MinInfoGain", "MaxStatsMemoryMB", "TreeNum", "Impurity", "FeatureSubsetStrategy", "Loss", "LearningDecay",
+            "Propagation", "GBTSampleWithReplacement", "Kernel", "Const", "Gamma", "EnableEarlyStop",
+            "ValidationTolerance", "MaxLeaves", "MaxBatchSplitSize" });
```
Comment (zhangpengshan) [inline src/main/java/ml/shifu/shifu/container/meta/MetaFactory.java]: One issue I left here, can we make such parameters not configured here but read from ModelConfigMeta.json?

File: src/main/java/ml/shifu/shifu/core/dtrain/gs/GridSearch.java [hunk 2]
```diff
@@ -282,6 +282,15 @@ private Object convertItemValue(Map<String, MetaItem> metaWarehouse, String item
         itemValueStr = itemValueStr.trim();
         if(itemMeta.getType().equals("text")) {
             return itemValueStr;
+        } else if(itemMeta.getType().equals("integer")) {
```
Comment (zhangpengshan) [inline src/main/java/ml/shifu/shifu/core/dtrain/gs/GridSearch.java]: itemMeta.getType().equals("integer") || itemMeta.getType().equals("int") I think is better to make it be stronger

File: src/main/java/ml/shifu/shifu/container/meta/MetaFactory.java (hunk 1 above)
Comment (junshiguo) [reply src/main/java/ml/shifu/shifu/container/meta/MetaFactory.java]: @zhangpengshan I think using key string starting with train#params# should cover all train params?

File: src/main/java/ml/shifu/shifu/core/dtrain/gs/GridSearch.java (hunk 2 above)
Comment (junshiguo) [reply src/main/java/ml/shifu/shifu/core/dtrain/gs/GridSearch.java]: @zhangpengshan will update it!

File: src/main/java/ml/shifu/shifu/container/meta/MetaFactory.java (hunk 1 above)
Comment (zhangpengshan) [reply src/main/java/ml/shifu/shifu/container/meta/MetaFactory.java]: that's very good idea, thanks:)

**Pull request merged**